components/popup/src/vwc-popup-base.ts

```diff
@@ -0,0 +1,198 @@
+import {
+	PropertyValues, html, LitElement, property, query, TemplateResult
+} from 'lit-element';
+import { ClassInfo, classMap } from 'lit-html/directives/class-map.js';
+import { nothing } from 'lit-html';
+import { computePosition, offset, shift, flip, arrow } from '@floating-ui/dom';
+import type { Placement, Strategy, Padding } from '@floating-ui/core';
+import '@vonage/vwc-elevation';
+
+export class VWCPopupBase extends LitElement {
+	@query('.popup') protected popupEl!: HTMLElement;
+	@query('.popup-arrow') protected arrowEl!: HTMLElement;
+	protected padding: Padding = 6;
+
+	/**
+	 * @prop open - indicates whether the popup is open
+	 * accepts boolean value
+	 * @public
+	 * */
+	@property({ type: Boolean, reflect: true })
+		open?: boolean= false;
+
+	/**
+	 * @prop anchor - the anchor of the popup
+	 * accepts Element
+	 * @public
+	 * */
+	@property({ type: Element, reflect: true })
+		anchor?: Element;
+
+	/**
+	 * @prop dismissible - adds close button to the popup
+	 * accepts boolean value
+	 * @public
+	 * */
+	@property({ type: Boolean, reflect: true })
+		dismissible?: boolean;
+
+	/**
+	 * @prop distance - distance offset
+	 * accepts number
+	 * @public
+	 * */
+	@property({ type: Number, reflect: true })
+		distance = 10;
+
+	/**
+	 * @prop corner - the placement of the popup
+	 * accepts  | 'top'
+				| 'top-start'
+				| 'top-end'
+				| 'right'
+				| 'right-start'
+				| 'right-end'
+				| 'bottom'
+				| 'bottom-start'
+				| 'bottom-end'
+				| 'left'
+				| 'left-start'
+				| 'left-end';
+	 * @public
+	 * */
+	@property({ type: String, reflect: true })
+		corner: Placement = 'left';
+
+	/**
+	 * @prop strategy - the position of the popup
+	 * accepts 'absolute' | 'fixed';
+	* @public
+	* */
+	@property({ type: String, reflect: true })
+		strategy: Strategy = 'absolute';
+
+	/**
+	 * @prop arrow - adds small triangle to indicate the trigger element
+	 * accepts boolean value
+	 * @public
+	 * */
+	@property({ type: Boolean, reflect: true })
+		arrow?: boolean;
+
+	/**
+	 * @prop alternate - set the color-scheme to dark
+	 * accepts boolean value
+	 * @public
+	 * */
+	@property({ type: Boolean, reflect: true })
+		alternate?: boolean;
+
+	protected override firstUpdated(changedProperties: PropertyValues): void {
+		super.firstUpdated(changedProperties);
+		this.open === true ? this.show() : this.hide();
+	}
+
+	protected override updated(changes: Map<string, boolean>): void {
+		super.updated(changes);
+		this.open === true ? this.show() : this.hide();
+	}
+
+	/**
+	 * Opens the popup
+	 * @public
+	 */
+	async show(): Promise<void> {
+		await this.updatePosition() ? (this.open = true) : this.hide();
+	}
+
+	/**
+	 * Closes the popup
+	 * @public
+	 */
+	hide(): void {
+		this.open = false;
+	}
+
+	/**
+	 * Updates popup position, if succeeded returns - true, if not - false
+	 * @public
+	 */
+	async updatePosition() {
```

Comment:
should this be called on resize and scroll. noticed the popup isn't aware of changes to viewport. might be heavy on performance